### ISSUE TITLE
[HOLD for payment 2022-12-20] [$1000] iOS - Attachments - Uploading image throws error "Unsupported PDF format"

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#12283]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to Settings > Camera > Format > ensure that "high-efficiency" is enabled
2. Open the iOS app
3. Visit any DM
4. Take a picture with the native iOS camera, ensuring that "live" photo is disabled
5. Tap the grey plus icon to the right of the chat input
6. Select a image from your camera roll (for iOS should be HEIC files by default)
7. Upload the image

## Expected Result:
The image should upload natively with no errors

## Actual Result:
RBR error appears, with the message "Unsupported PDF format"

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- iOS

**Version Number:** 1.2.21.4

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** 

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/198847076-65bd917b-150a-41cb-8da0-a5e5bb649a6a.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0156f41c36e242a9d2</li>
        <li>Upwork Job ID: 1602341181291810816
<li>Last Price Increase: 2022-12-12</li></li>
    </ul>
</details>

**Comment by mateocole** [comment]
having engineering take a look

**Comment by arosiclair** [comment]
I don't have a physical iPhone to repro this, but this does look to be fixable by a contributor. Marking external.

**Comment by JmillsExpensify** [comment]
@dhairyasenjaliya Thanks for flagging. I commented in the linked issue immediately above this post.

**Comment by Santhosh-Sellavel** [comment]
@mmarceau Your proposal looks let's make it a new method `isAttachmentDownloadable` and add a checks with relevant comments for understanding!

**Comment by Santhosh-Sellavel** [comment]
@kbecciv 
> Try to upload unsupported file type and long type on the error message--> Download

Can you send a screen recording for this step and a file please!

**Comment by kbecciv** [comment]
@Santhosh-Sellavel Please attached video, you need to turn off live mode when taking a picture before attaching in NewDot app.

https://user-images.githubusercontent.com/93399543/200347442-09252155-cace-4fb5-b1d1-ec93b01fa217.mp4

**Comment by JmillsExpensify** [comment]
This issue still seems like a duplicate of the crash that we're also discussing in https://github.com/Expensify/App/issues/10907. Right?

**Comment by JmillsExpensify** [comment]
@Gonals Do you mind chiming in on my last question? Looking at the reproduction steps, I'm failing to see the difference.

**Comment by Santhosh-Sellavel** [comment]
@JmillsExpensify Uploading is a duplicate of the other issue, but this here https://github.com/Expensify/App/issues/12283#issuecomment-1305774920, where we show a preview of the image they selected and followed by an error on sending is weird, I think we should do something about it

**Comment by Gonals** [comment]
Yes, I think @Santhosh-Sellavel. We seem to be dealing with two different problems here:
1. Error after showing the image preview
2. Crash downloading the attachment after failure (duplicate)

**Comment by JmillsExpensify** [comment]
Nice, yeah agreed. I think one is also a duplicate as well though. Let me try to find this issue, this is related to changes to the attachment flow, where everything is running through Web-PDFs.

**Comment by trjExpensify** [comment]
>Error after showing the image preview - @Gonals 
>I think one is also a duplicate as well though. Let me try to find this issue, this is related to changes to the attachment flow, where everything is running through Web-PDFs. - @JmillsExpensify 

So this issue can be closed can it, if it's being taken care of elsewhere?

**Comment by JmillsExpensify** [comment]
Updated this issue to remove the duplicate discussion around the crash. That's handled in https://github.com/Expensify/App/issues/10907 moving forward. This issue is now focused on the on the "Unsupported PDF format" RBR error message moving forward.

**Comment by JmillsExpensify** [comment]
Separately, @youssef-lr still would like confirmation if you are aware of this issue or working on any PRs elsewhere that resolve this?

**Comment by Santhosh-Sellavel** [comment]
> Updated this issue to remove the duplicate discussion around the crash. That's handled in #10907 moving forward. This issue is now focused on the on the "Unsupported PDF format" RBR error message.

Thanks, @JmillsExpensify, now its more clear!

**Comment by JmillsExpensify** [comment]
Also an update on this issue: It's pretty tricky. I tested two flows, and received a different result each time. Basically, if you follow the testing flow on mWeb, you receive no error. It's also possible to see the (HEIC) attachment after upload. However, if you test on native iOS, then the reproduction steps above produce the error. That makes me think that we're doing some combination of file processing in one flow that doesn't hit the error, whereas the native flow is treating the file differently and so, triggers the error.

**Comment by JmillsExpensify** [comment]
Now that we're aligned on the issue/next steps, I'm doubling the price of this issue!

**Comment by JmillsExpensify** [comment]
Still waiting on proposals.

**Comment by puneetlath** [comment]
@Gonals To help us clear out the large backlog of /App bugs, we're putting the spotlight every bug in the repo already than 4 weeks old. To help unblock the roadmap and get our bug pipeline back in equilibrium, can you:  
 
- Decide whether any proposals currently meet our guidelines and can be approved as-is 
- For any that can't, please take this issue internal and treat it as one of your highest priorities 
- If you have any questions, don't hesitate to start a discussion in #bug-zero

**Comment by Gonals** [comment]
Hey @puneetlath! So we don't plan to increase the price of this? It got doubled just a few days ago

**Comment by puneetlath** [comment]
Ah actually, my mistake. This isn't 4+ weeks old yet, so it doesn't **have** to be taken internal. I think it might be worth you asking if another engineer wants to trade you for this issue though, since you don't have iOS.

**Comment by s77rt** [comment]
### Proposal
```diff
diff --git a/src/components/AttachmentPicker/index.native.js b/src/components/AttachmentPicker/index.native.js
index 418be07b8..aa965e5e7 100644
--- a/src/components/AttachmentPicker/index.native.js
+++ b/src/components/AttachmentPicker/index.native.js
@@ -34,6 +34,7 @@ const imagePickerOptions = {
     saveToPhotos: false,
     selectionLimit: 1,
     includeExtra: false,
+    quality: 0.95,
 };
 
 /**
```

### Details
My proposal is more like a workaround, the bug is in react-native-image-picker 
I think we can proceed with the workaround without waiting for the react-native-image-picker.

**Comment by Santhosh-Sellavel** [comment]
Strictly No workarounds, if the issue exists at the library, it's better to address it there! If you have a solution to solve please post it here, i.e at the library itself!

**Comment by s77rt** [comment]
@Santhosh-Sellavel I have submitted a bug report already.
I have found the root cause, proposed a solution for the library and proposed a workaround here.

**Comment by Santhosh-Sellavel** [comment]
> @Santhosh-Sellavel I have submitted a bug report already. I have found the root cause, proposed a solution for the library, and proposed a workaround here.

If you have found the root cause then why was the workaround submitted here?

**Comment by s77rt** [comment]
@Santhosh-Sellavel 
From Expensify App perspective `react-native-image-picker` is the root cause. It's related to the `quality` prop. I know that the issue exist for setting the value equal to `1`.
From `react-native-image-picker` perspective I don't know the root cause, I just submitted a suggestion to where to start looking in the code.

**Comment by JmillsExpensify** [comment]
While the conversation above continues, doubling the price of this issue once more! We're now at $1,000 💰

**Comment by s77rt** [comment]
The issue seems to be a regression from https://github.com/react-native-image-picker/react-native-image-picker/pull/2006
I have asked if this can be reverted, then it should be sorted out, except that the returned file will seem to be double but that's just due to the conversion from heic to jpeg I suppose. We may also want to keep the quality at `0.95`

**Comment by mountiny** [comment]
> except that the returned file will seem to be double but that's just due to the conversion from heic to jpeg I suppose

I dont think reverting that fix is great, that was introduced to fix another issue. Doubling the size is not what we want and I believe it doubles the size also for images which are jpeg in the gallery

**Comment by s77rt** [comment]
@mountiny What I'm saying is that maybe the PR didn't fix any issue in the first place and the returned size was actually the correct size.
I have downloaded the image that seems to have the size doubled, and the size was in fact correct.
Uploaded image at quality 1 => size 9.6MB (this seems to be double but it's correct) => downloaded the image and the size is in fact 9.6MB ==> Correct
Uploaded image at quality 0.95 => size 5.1MB => downloaded the image and the size is 5.1MB ==> Correct

**Comment by Santhosh-Sellavel** [comment]
> @mountiny What I'm saying is that maybe the PR didn't fix any issue in the first place and the returned size was actually the correct size.@s77rt 
Which platform did you check?

**Comment by s77rt** [comment]
@Santhosh-Sellavel Your question is not quite clear, check what exactly?

**Comment by s77rt** [comment]
@Santhosh-Sellavel Ah, you quoted the wrong part, your question was out of context.
1. I have modified the code (basically a revert) so I can get it working.
2. Send the image using quality 1 (on iOS)
3. Downloaded the image that I sent (on Linux, but this does not matter, it just happen to be Linux)
4. Verified that the reported size on iOS 9.6MB is indeed the size of the photo that I downloaded

**Comment by s77rt** [comment]
Here is the image, you can download it and you will see it's indeed 9.6MB. There was nothing wrong it just happen that the conversion at quality 1 is so inefficient that the size seems to be double. 

![w_8361c37cbbfcef589f6fc29b76d5de220586b302](https://user-images.githubusercontent.com/16493223/203803290-f0e2eeef-5631-4e26-acc3-f7c22ab58aa3.jpg)

**Comment by Santhosh-Sellavel** [comment]
Send the image using iOS and download it on any platforms size will change.

**Comment by s77rt** [comment]
> Send the image using iOS and download it on any platforms size will change.

Isn't that what i just said i did?

**Comment by mountiny** [comment]
So with the revert of the PR you mention did not solve any issue, if you upload JPG file and then download it again, is the size not doubled? Are you saying this only happens for HEIC images because of the conversion?

**Comment by s77rt** [comment]
@mountiny Why do you think the size will double? it's just impossible to get the size doubled on download. You download what you upload.

Let me make it clear:
If you upload the file at quality 1 (original size: 5.1MB), you will be uploading the converted file and since the conversion is not that good, the file size will be much larger than what you'd expect (9.6MB).
The reported size by react-native-image-picker is correct (9.6MB).

If you download it, you will of course get the same amount of data that has been uploaded, thus the same file size (9.6MB).

**Comment by Santhosh-Sellavel** [comment]
5.1 MB shouldn't become 9.6 MB that's the problem that is addressed in the PR https://github.com/react-native-image-picker/react-native-image-picker/pull/2006

**Comment by s77rt** [comment]
@Santhosh-Sellavel it's not that 5.1MB is magically set to 9.6MB it's the returned data that's much larger as I said due to the conversion.
With the PR, the returned size at quality 1 is about 2.6MB which is totally wrong because at quality 0.95 it's 5.1MB and that does not make sense.

**Comment by mountiny** [comment]
> it's the returned data that's much larger as I said due to the conversion.

Yes, the scenario I am asking about is if we upload JPG file, then there should be no conversion, why would we convert JPG. I understand the HEIC will get converted because _Apple_ but what about normal formats like PNG or JPG, those should not change type and hence quality 1 should keep the same filesize when uploaded @s77rt

**Comment by s77rt** [comment]
@mountiny Ah okay, now I see the problem, every file is being converted but the file size correct.
Will have to look into this more

**Comment by s77rt** [comment]
Okay so far I found the root cause. I can apply this fix:
1. Process other files just normally (jpg, png, ...) (without extra conversion / doubling the size)
2. Convert HEIC files to jpeg

However the converted HEIC file size is almost the double (5.1 MB TO 9.6MB)
I think this can be improved, because browsers can select HEIC files and convert them without that much diff, maybe they don't set the quality to 1?

Will update you shortly

**Comment by s77rt** [comment]
### Proposal (Updated)
```diff
diff --git a/node_modules/react-native-image-picker/ios/ImagePickerUtils.m b/node_modules/react-native-image-picker/ios/ImagePickerUtils.m
index 620ddc9..3afe9d6 100644
--- a/node_modules/react-native-image-picker/ios/ImagePickerUtils.m
+++ b/node_modules/react-native-image-picker/ios/ImagePickerUtils.m
@@ -58,7 +58,7 @@
         configuration = [[PHPickerConfiguration alloc] init];
     }
     
-    configuration.preferredAssetRepresentationMode = PHPickerConfigurationAssetRepresentationModeCurrent;
+    configuration.preferredAssetRepresentationMode = PHPickerConfigurationAssetRepresentationModeAutomatic;
     configuration.selectionLimit = [options[@"selectionLimit"] integerValue];
 
     if ([options[@"mediaType"] isEqualToString:@"video"]) {
```

### Update
Fixed the issue on the react-native-image-picker level

### Details
The problem is in PHPicker. The selected file per default will have it's original format, in this case it's heic.
And we are uploading this file directly to the server which is not supported.

In my proposal I changed the `preferredAssetRepresentationMode` from [PHPickerConfigurationAssetRepresentationModeCurrent](https://developer.apple.com/documentation/photokit/phpickerconfigurationassetrepresentationmode/phpickerconfigurationassetrepresentationmodecurrent?language=objc) to [PHPickerConfigurationAssetRepresentationModeAutomatic](https://developer.apple.com/documentation/photokit/phpickerconfigurationassetrepresentationmode/phpickerconfigurationassetrepresentationmodeautomatic?language=objc) This setting will automatically select the appropriate format; for heic files we will get jpeg. This is also the behaviour of `UIImagePickerController` (prior iOS 14) so this issue only effect iOS 14+ users due to said problem

If it is not clear, the library uses two image pickers. `UIImagePickerController` and `PHPickerViewController`. Uses `PHPickerViewController` if available (which is available on iOS 14+). else fallback to `UIImagePickerController`.
In my proposal I'm matching the behaviour of `PHPickerViewController` to `UIImagePickerController`.


PS: This needs to be tested on a real device, iOS simulator does not properly support phpicker heic to jpeg format.

___

Additional info and correction about the file size:
2.8MB is the file size of the heic file
5.1MB should be the expected file size of jpeg (this will be supplied directly from iOS, we won't handle any conversion from our side)
9.6MB is the size of the converted heic file to jpeg using `UIImageJPEGRepresentation` (very inefficient)

Meaning, with my proposal you should expect the file size of the image (pink and yellow flowers) to be 5.1MB

**Comment by JmillsExpensify** [comment]
General update: We're still working through the issue/cause and related proposals. Latest proposal is immediately above for review.

**Comment by Santhosh-Sellavel** [comment]
@s77rt Thanks for the proposal.

I'm not sure this is the right path

> In my proposal I changed the preferredAssetRepresentationMode from [PHPickerConfigurationAssetRepresentationModeCurrent](https://developer.apple.com/documentation/photokit/phpickerconfigurationassetrepresentationmode/phpickerconfigurationassetrepresentationmodecurrent?language=objc) to [PHPickerConfigurationAssetRepresentationModeAutomatic](https://developer.apple.com/documentation/photokit/phpickerconfigurationassetrepresentationmode/phpickerconfigurationassetrepresentationmodeautomatic?language=objc) This setting will automatically select the appropriate format; for heic files we will get jpeg. This is also the behaviour of UIImagePickerController (prior iOS 14) so this issue only effect iOS 14+ users due to said problem

Can you elaborate more on how this address the issue? What really happens when switching?

Have you submitted the PR on react-native-image-picker repo?  If not, please raise a PR to let's hear from them too!

**Comment by s77rt** [comment]
@Santhosh-Sellavel 
I think I have already stated the reason for switching. Which is to use the JPEG format and not the HEIC format.

It's either:
1. Use HEIC then convert it to JPEG using `UIImageJPEGRepresentation`
2. Use directly the provided JPEG format

The first approach is inefficient and results in a bigger file unless we carefully choose the quality.
The second approach is better since the conversion is done by iOS and the quality is (I suppose) chosen automatically.

I have proposed the solution on the created issue https://github.com/react-native-image-picker/react-native-image-picker/issues/2052
I have also proposed an alternative solution that follows the same principle

___

Summary:
We are getting the HEIC file and uploading it to the server which is unsupported thus the error.
We should be getting the JPEG file and upload it instead. Changing `preferredAssetRepresentationMode` will get us the JPEG file directly without the need of an explicit conversion.

**Comment by Santhosh-Sellavel** [comment]
@s77rt  

```
It's either:

Use HEIC then convert it to JPEG using UIImageJPEGRepresentation
Use directly the provided JPEG format
```
Can you share any supporting doc for this?

**Comment by s77rt** [comment]
@Santhosh-Sellavel 
For what exactly? the statement is clear, does not need an external docs.

PHPickerConfigurationAssetRepresentationModeCurrent will select the original format for every file.
PHPickerConfigurationAssetRepresentationModeAutomatic will select the format automatically
PHPickerConfigurationAssetRepresentationModeCompatible will select the most compatible format

You don't want to use PHPickerConfigurationAssetRepresentationModeCurrent because it will give you the HEIC file format.

You have to convert it unless you want to support HEIC files internally or make the conversion server-side.
or
Just get the JPEG format in the first place and upload it to the server

I'm not sure what part requires docs

**Comment by Santhosh-Sellavel** [comment]
> @Santhosh-Sellavel 
> For what exactly? the statement is clear, does not need an external docs.
> 
> PHPickerConfigurationAssetRepresentationModeCurrent will select the original format for every file.
> PHPickerConfigurationAssetRepresentationModeAutomatic will select the format automatically
> PHPickerConfigurationAssetRepresentationModeCompatible will select the most compatible format
> 
> You don't want to use PHPickerConfigurationAssetRepresentationModeCurrent because it will give you the HEIC file format.
> 
> You have to convert it unless you want to support HEIC files internally or make the conversion server-side.
> or
> Just get the JPEG format in the first place and upload it to the server
> 
> I'm not sure what part requires docs 

From where we can learn this?

**Comment by s77rt** [comment]
I have provided links already for the preferred selection modes.
https://developer.apple.com/documentation/photokit/phpickerconfigurationassetrepresentationmode/phpickerconfigurationassetrepresentationmodeautomatic?language=objc
https://developer.apple.com/documentation/photokit/phpickerconfigurationassetrepresentationmode/phpickerconfigurationassetrepresentationmodecurrent?language=objc

**Comment by mountiny** [comment]
@Gonals @Santhosh-Sellavel what do you think here, bumping this issue.

**Comment by Santhosh-Sellavel** [comment]
@mountiny Looks good to me so far, yet to test this one on a real device.

Meanwhile please @s77rt, please bump the maintainer of the library. I would suggest opening up a PR there instead of an issue report, thanks!

**Comment by s77rt** [comment]
@Santhosh-Sellavel The library does not seem very active it may be better to apply a patch or fork the repo if it took to long
Let's test the proposal first

**Comment by Santhosh-Sellavel** [comment]
Let's try it out first if unresponsive for over a week, let's do it on a fork!

**Comment by JmillsExpensify** [comment]
That plan sounds fine to me, though @Gonals please weigh in tomorrow when you're back on.

**Comment by s77rt** [comment]
Here is the PR for reference
https://github.com/react-native-image-picker/react-native-image-picker/pull/2056

**Comment by Gonals** [comment]
That sounds good to me too, but let's not wait a week. If we don't hear back in a couple of days, let's go ahead and create a fork

**Comment by mountiny** [comment]
Forking is quite annoying to maintain, this maintainer has helped us before, I will ping them

**Comment by JmillsExpensify** [comment]
Agreed, very non-ideal.

**Comment by s77rt** [comment]
The PR has been merged
Now we need a tester

**Comment by mountiny** [comment]
We will sponsor the library as we had couple of PRs in there. 

@Santhosh-Sellavel do you have Android and iphone to test this on real device once we will be in a review?

@s77rt We need a PR, can you create it with the new version and test also for the regressions/ previous issues we had in the library

**Comment by s77rt** [comment]
Applied in Upwork (+50% bonus) in case this fixed the issue
[PR](https://github.com/Expensify/App/pull/13246) is ready

**Comment by Santhosh-Sellavel** [comment]
PR is failing for android, so I went ahead and tested on other platforms, seems we don't allow this format on the web/desktop either.

https://user-images.githubusercontent.com/85645967/205384598-bfa7454f-2e38-4f1e-bd70-ba1f29c8b0d7.mov

**Comment by s77rt** [comment]
@Santhosh-Sellavel You are testing it the wrong way.
You are not supposed to select a raw .heic file that way.
What I meant is to select an image that is internally stored as heic.

You are only supposed to test on iOS

**Comment by mountiny** [comment]
@s77rt @Santhosh-Sellavel The issue is mentioned as iOS but we should aim to solve this for Android too, the usecase is edge case so I would say we should merge the PR which fixed iOS where this is a blocker to sending images and then we should create a follow up PR to fix Android.

This issue mentions iOS, but I would argue in this case it is obvious and makes sense to fix any native mobile too.

@Santhosh-Sellavel the issue shown above is not related to this change. cc @youssef-lr afaik you have worked on accepting the heic on web, do you know what is the state of that?

**Comment by s77rt** [comment]
@mountiny I don't see an issue with Android. Can you upload a video?

**Comment by mountiny** [comment]
Just coming from the video @Santhosh-Sellavel has shared [here](https://github.com/Expensify/App/pull/13246#issuecomment-1335844139), it is using jpeg image, this should be working on `main`, should it not?

**Comment by JmillsExpensify** [comment]
> The issue is mentioned as iOS but we should aim to solve this for Android too, the usecase is edge case so I would say we should merge the PR which fixed iOS where this is a blocker to sending images and then we should create a follow up PR to fix Android.

I agree with this.

**Comment by JmillsExpensify** [comment]
@Gonals @Santhosh-Sellavel What are our next steps for this issue? Any movement on the Android consideration?

**Comment by Santhosh-Sellavel** [comment]
Thanks for the bump! @Gonals Please could you add help wanted label again!

**Comment by s77rt** [comment]
@JmillsExpensify @Santhosh-Sellavel @Gonals Shouldn't we handle the android case in it's own separated issue? as a feature request?

**Comment by Gonals** [comment]
Since the issue existed both in Android and iOS to begin with, and the whole conversation is here, I think it makes sense to keep using this issue

**Comment by s77rt** [comment]
@Gonals What about the compensation is it 2x $1000 in this case?
Are we expecting two contributors and two PRs for one issue?

**Comment by joekaufmanexpensify** [comment]
Discussing next steps in [Slack. ](https://expensify.slack.com/archives/C02NK2DQWUX/p1670874971108159)

**Comment by mountiny** [comment]
Readding the `External` label will create a new job in Upwork @Gonals just noting. I think you might have wanted to use `Bug` label instead :) 

@s77rt It will be a new job and I think it might have to be internal job after all as it might need access to API.

**Comment by JmillsExpensify** [comment]
I'm returning on the 20th, so I'll circle back then for the regression test and payment.

**Comment by joekaufmanexpensify** [comment]
It sounds like @JmillsExpensify has this one handled for BZ, so I'm going to un-assign, as it looks like I was just assigned to help cover OOO. LMK if there's anything I can do to help though!

**Comment by JmillsExpensify** [comment]
Thanks @joekaufmanexpensify i really appreciate the help while OOO. I'll circle back to this one in the coming hours to close the loop. 🙌

**Comment by JmillsExpensify** [comment]
Ok I'm circling back on this one. We have a decision to make here before going through the Upwork hire process and issuing payment, specifically:

@mountiny and @Santhosh-Sellavel I think we need to address @s77rt's [question in the PR](https://github.com/Expensify/App/pull/13246#issuecomment-1339326216) as to whether the 50% bonus applies. I think that his perspective is that it _could_ have been merged sooner but wasn't due to the discussion that occurred during the testing process. However, I think another perspective is that it wasn't, and due to the through testing by C+ to make sure we got the scope of the issue right, as well as didn't introduce regressions, it took longer. I think I am more amenable to the latter argument. Please weigh in regardless and let's treat this objectively and fairly thanks!

**Comment by JmillsExpensify** [comment]
Additionally, @Santhosh-Sellavel (@mountiny - I'm pulling you in given that Alberto is OOO until the New Year and you have context from the PR), do you mind handling the first three items on the Bug Zero checklist? Given the upstream fix and considerations, I think we should be good on those, though let's confirm either way. Then I can circle back on the regression test pulling from the linked PR.

**Comment by Gonals** [comment]
Wait, I'm OOO un til New Year? When did that happen? I'm only OOO some days! 😆 

@JmillsExpensify, I'm with you on this one. I don't think I've seen other cases where the bonus was requested when the PR took longer to merge, but, in my opinion, discussions are part of the PR process, and this was not something that got "stuck" without movement for a while. There was back and forth and that's part of the process and, sometimes, PRs take longer for that 🤷 .

**Comment by mountiny** [comment]
I agree with both of you, in this particular case I think the delay was not due to the lack of response from us but more like a normal discussion about the outcome of the PR. 

@Gonals will you be able to handle the checklist since you are assigned here and apparently not ooo 😂

**Comment by Gonals** [comment]
Sure thing!

**Comment by JmillsExpensify** [comment]
Thanks for weighing in! I'll move forward with the following payments:

 - Contributor: @s77rt $1,000
 - Contributor+: @Santhosh-Sellavel $1,000

Also @s77rt while I appreciate you want to be optimistic and submit for the bonus when the PR is up, I did want to reiterate that this is not our process per our Contributing.md. The bonus is applied and eligible when the PR is merged, not when the PR is up for approval. Let's please standardize on not asking for the bonus ahead of time moving forward. Thank you.

**Comment by JmillsExpensify** [comment]
Quick update: Contract extended to @s77rt and @Santhosh-Sellavel invited since he wasn't on the job yet.

**Comment by JmillsExpensify** [comment]
All contributors paid out. Closing this issue and I'll link the regression test very shortly.

**Comment by JmillsExpensify** [comment]
Closed the loop on the regression test. It is linked immediately above!
